fabric-entity-events-v1/src/testmod/java/net/fabricmc/fabric/test/entity/event/EntityEventTests.java

```diff
@@ -80,6 +80,20 @@ public void onInitialize() {
 			LOGGER.info("Respawned {}, [{}, {}]", oldPlayer.getGameProfile().getName(), oldPlayer.getWorld().getRegistryKey().getValue(), newPlayer.getWorld().getRegistryKey().getValue());
 		});
 
+        ServerPlayerEvents.ALLOW_TELEPORT.register((player, pos) -> {
+            if(pos.getY() < -256)
```

Comment:
Might be better to prevent the player from teleporting if they are holding a specific item, this way its easier to test portals.

Reply:
Using portals to travel across dimensions is different from teleporting with coordinates or between players, making it difficult to unify the behaviors. AllowTeleport and AfterTeleport is only suitable for the latter, whether using the tp command or a map mod.